tests/kernel/pipe/pipe/src/test_pipe.c

```diff
@@ -874,3 +883,41 @@ void test_pipe_put_min_xfer(void)
 	zassert_equal(res, 0, "did not write min_xfer");
 	zassert_true(bytes_written >= 1, "did not write min_xfer");
 }
+
+/**
+ * @brief Test defining and initializing pipes at run time
+ * @details Check if the kernel provided a mechanism for defining and
+ * initializing pipes at run time.
+ * - Define and initialize a pipe at run time
+ * - Using this pipe to transfer data and verify it
+ * @ingroup kernel_pipe_tests
+ * @see k_pipe_init()
+ */
+void test_pipe_define_at_runtime(void)
+{
+	unsigned char __aligned(4) ring_buffer[PIPE_SIZE];
```

Comment:
What does this need alignment?  If it's because the test code is using the buffer as an array of other types, then you should declare the memory as an array of whatever the natural type is.  If it's because the pipe code is doing it and hitting bus errors at runtime, we have a bug in the pipe layer (which is supposed to be byte-aligned only).  If it's neither (and I think it's neither), just remove the alignment restriction.

Reply:
o,  yes,  needn't memory alignment here,  Just moved,  thanks for reminding. :)